src/keyring/KeyringController.test.ts

```diff
@@ -1094,4 +1096,264 @@ describe('KeyringController', () => {
       expect(cancelSignRequestStub.called).toBe(true);
     });
   });
+
+  describe('Ledger Keyring', () => {
+    let ledgerKeyring: LedgerKeyring;
+    preferences = new PreferencesController();
+
+    beforeEach(async () => {
+      keyringController = new KeyringController(
+        {
+          setAccountLabel: preferences.setAccountLabel.bind(preferences),
+          removeIdentity: preferences.removeIdentity.bind(preferences),
+          syncIdentities: preferences.syncIdentities.bind(preferences),
+          updateIdentities: preferences.updateIdentities.bind(preferences),
+          setSelectedAddress: preferences.setSelectedAddress.bind(preferences),
+        },
+        baseConfig,
+      );
+
+      await keyringController.createNewVaultAndKeychain(password);
+      ledgerKeyring = await keyringController.getLedgerKeyring();
+
+      ledgerKeyring.setApp({
+        signTransaction: sinon.stub(),
+        getAddress: sinon.stub().resolves({
+          address: '0xe908e4378431418759b4f87b4bf7966e8aaa5cf2',
+        }),
+        signEIP712HashedMessage: sinon.stub(),
+        signPersonalMessage: sinon.stub(),
+      });
+
+      await keyringController.unlockLedgerDefaultAccount();
+    });
+
+    it('should unlock default account from Ledger', async () => {
+      const defaultAccount =
+        await keyringController.unlockLedgerDefaultAccount();
```

Comment:
Seems like this already happens as the last step of the setup — maybe we should assign this variable in the setup instead?

Reply:
[dccbf68](https://github.com/MetaMask/controllers/pull/862/commits/dccbf68806ad0e0ac7e54c549eacdc02c687180d)